docs/development.md

```diff
@@ -60,4 +60,29 @@ To generate `zz_generated.deepcopy.go` & `zz_generated.swagger_doc_generated.go`
 3. Run `make update-scripts` to see `zz_generated.deepcopy.go` & `zz_generated.swagger_doc_generated.go` generated in your API directory, and along with some related scripts generated in `client` directory. If `zz_generated.deepcopy.go` is not generated properly, please run the command in a container with `RUNTIME=docker make update-with-container` (if you are using `podman`, set `RUNTIME=podman`).
 
 ## Verify
-Before you commit you changes, please run `make verify` locally to make sure you have generated all required files.
\ No newline at end of file
+Before you commit you changes, please run `make verify` locally to make sure you have generated all required files.
+
+## API upgrade flow
+When introducing a new version API, you need to deprecate the old version API and then remove it. 
+Below is the suggested API migration flow, the example code is to migrate `ManagedClusterSet` API from v1beta1 to v1beta2.
+
+### Relase N (add new version and depracated old version)
+- Add new version v1beta2 API and mark v1beta1 as deprecated. For example: 
```

Comment:
My understanding is we mark it deprecated when we change the stored version?

Reply:
Sounds makes sense. 

In our previous releases, we tended to add new API and mark the old version deprecated first, letting the user know it's deprecated for at least 1 release and then start migration in the next release. 

Now we have some experience in migration, it's okay to mark it deprecated when changing the stored version in release N+1.

Reply:
have updated the doc, also with some tips on how to set the served and storage version.